src/ChannelServer/Network/Handlers/Parties.cs

```diff
@@ -0,0 +1,308 @@
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+            
+            if (creature == null)
+                return;
+
+            if (creature.IsInParty)
+                return;
+
+            // Needs check like the below, but do homesteads count as dynamic regions? I know you can't make a party in the shadow realm (you can in dungeons), but are they the only places? Need to look into that.
+            //if (!creature.Region.IsDynamic)          
+            {
+                var party = creature.Party;
+
+                party.CreateParty(creature, packet);
+
+                Send.CreatePartyR(creature);
+
+                party.Open();
+                return;
+            }
+            //else
+            //Send.CreatePartyR(creature);
+        }
+
+        [PacketHandler(Op.PartyJoin)]
+        public void PartyJoin(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+
+            if (creature == null)
+                return;
+
+            if (creature.IsInParty)
+                return;
+
+            var partyLeader = ChannelServer.Instance.World.GetCreature(packet.GetLong());
+
+            if (!partyLeader.IsInParty)
+                return;
+            if (partyLeader.Party.Leader.EntityId != partyLeader.EntityId)
+                return;
+
+            var party = partyLeader.Party;
+            var password = packet.GetString();
+
+            if(party.TotalMembers >= party.MaxSize)
+            {
+                Send.PartyJoinR(creature, World.PartyJoinResult.Full);
+                return;
+            }
+
+            if (party.Password != password)
+            {
+                Send.PartyJoinR(creature, World.PartyJoinResult.WrongPass);
+                return;
+            }
+
+            party.AddMember(creature);
+            Send.PartyJoinUpdateMembers(creature);
+
+            if (party.IsOpen)
+                Send.PartyMemberWantedRefresh(party);
+
+            Send.PartyJoinR(creature, World.PartyJoinResult.Success);
+            Send.PartyCreateUpdate(creature);
+        }
+
+        [PacketHandler(Op.PartyLeave)]
+        public void LeaveParty(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+
+            if (creature == null)
+            {
+                Send.PartyLeaveR(creature, false);
+                return;
+            }
+
+            if (!creature.IsInParty)
+            {
+                Send.PartyLeaveR(creature, false);
+                return;
+            }
+
+            var party = creature.Party;
+
+            // TODO: Check if there are other regions or situations in which you cannot leave a party.
+            var canLeave = !creature.Region.IsDynamic;
+
+            if (canLeave)
+            {
+                creature.Party.RemoveMember(creature);
+
+
+                if (party.TotalMembers > 0)
```

Comment:
This should probably be done in `RemoveMember`, what if you remove a member from somewhere else? You would have to duplicate the code.